src/main/java/MainWindow.java

```diff
@@ -0,0 +1,84 @@
+import alice.Alice;
+import alice.command.Parser;
+import alice.task.TaskList;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+
+import java.io.File;
+import java.util.ArrayList;
+
+import static javafx.application.Application.launch;
+
+/**
+ * Controller for MainWindow. Provides the layout for the other controls.
+ */
+public class MainWindow extends AnchorPane {
+	@FXML
+	private ScrollPane scrollPane;
+	@FXML
+	private VBox dialogContainer;
+	@FXML
+	private TextField userInput;
+	@FXML
+	private Button sendButton;
+
+	private Alice alice;
+
+	private Image userImage = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+	private Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+
+	@FXML
+	public void initialize() {
+		scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+	}
+
+	private static String getDataPath() {
+		if (Storage.OS.contains("win")) {
+			return String.join(File.separator, System.getenv("USERPROFILE"),
```

Comment:
I like the way you separated the code over here! It makes it easier to read the long sentence.